src/cast_core/include/cast_core/actors/SamplingLoader.hpp

```diff
@@ -36,21 +81,27 @@ namespace genny::actor {
 class SamplingLoader : public Actor {
 
 public:
-    explicit SamplingLoader(ActorContext& context);
+    SamplingLoader(ActorContext& context,
+                   std::string dbName,
+                   std::string collectionName,
+                   std::shared_ptr<DeferredSample> deferredSamplingLoader);
     ~SamplingLoader() override = default;
 
     static std::string_view defaultName() {
         return "SamplingLoader";
     }
     void run() override;
 
+
 private:
     /** @private */
     struct PhaseConfig;
 
     metrics::Operation _totalBulkLoad;
     metrics::Operation _individualBulkLoad;
     mongocxx::pool::entry _client;
+    mongocxx::collection _collection;
```

Comment:
Intuitively we wouldn't need both a client and a collection since a collection comes from its client. Consider a comment here explaining why both are necessary.

Reply:
I think that a collection can't exist without it's client still being around in scope - so it's just convenience to have the collection member variable to save and use directly.